src/bootstrap/builder.rs

```diff
@@ -1252,9 +1252,19 @@ mod tests;
 struct Rustflags(String);
 
 impl Rustflags {
-    fn new() -> Rustflags {
+    fn new(target: &str, cmd: &mut Command) -> Rustflags {
         let mut ret = Rustflags(String::new());
+
+        // Inherit `RUSTFLAGS` by default
         ret.env("RUSTFLAGS");
+
+        // ... and also handle target-specific env RUSTFLAGS if they're
+        // configured. If this is configured we also remove it from the
+        // environment because Cargo will prefer it over RUSTFLAGS.
```

Comment:
Hm, my reading of Cargo code was rather the opposite -- Cargo will prefer RUSTFLAGS over the env-specific flags. But otherwise, this seems correct (and resolves an earlier comment).

Reply:
Yeah the purpose of this is to delete the target-specific env var so cargo instead uses `RUSTFLAGS`, which has the semantics we want here.

Reply:
Sure, I meant that I believe " environment because Cargo will prefer it over RUSTFLAGS" is not true, Cargo will prefer RUSTFLAGS over the target-specific environment variable.